# lowlevel controller
llc_config_file:
  /root/code/go2_ws/src/unitree_lowlevel/config/go2.yaml

# TorchScript policy for C++ inference
policy:
  backend: libtorch
  # Change to your exported TorchScript model path (.pt or .ts)
  # model_path: /root/code/go2_ws/src/legged_rl_deploy/policies/go2_trot.pt
  model_path: /root/code/go2_ws/src/legged_rl_deploy/policies/go2_velocity_unitree.pt
  obs_dim: 45
  act_dim: 12
  policy_dt: 0.02
  joint_ids_map: [3, 0, 9, 6, 4, 1, 10, 7, 5, 2, 11, 8]
  stiffness: [25.0, 25.0, 25.0, 25.0, 25.0, 25.0, 25.0, 25.0, 25.0, 25.0, 25.0, 25.0]
  damping: [0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5]
  commands: 
    base_velocity:
      process_order: [clip]
      min: [-1.0, -0.4, -1.0]
      max: [1.0, 0.4, 1.0]
  actions:
    JointPositionAction:
      process_order: [scale, offset, clip]
      min: -100.0
      max: 100.0
      scale: 0.25
      offset: [0.1, -0.1, 0.1, -0.1, 0.8, 0.8, 1.0, 1.0, -1.5, -1.5, -1.5, -1.5]
  observations:
    frame_stack: 1
    terms:
      base_ang_vel:
        process_order: [clip, scale]
        min: -100.0
        max: 100.0
        scale: [0.2, 0.2, 0.2]
      projected_gravity:
        process_order: [clip, scale]
        min: -100.0
        max: 100.0
        scale: [1.0, 1.0, 1.0]
      velocity_commands:
        process_order: [clip, scale]
        min: -100.0
        max: 100.0
        scale: [1.0, 1.0, 1.0]
      joint_pos:
        process_order: [offset, clip, scale]
        min: -100.0
        max: 100.0
        scale: 1.0
        offset: [-0.1, 0.1, -0.1, 0.1, -0.8, -0.8, -1.0, -1.0, 1.5, 1.5, 1.5, 1.5]
      joint_vel:
        process_order: [clip, scale]
        min: -100.0
        max: 100.0
        scale: [0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05]
      last_action:
